homeassistant/components/homekit/type_sensors.py

```diff
@@ -75,3 +75,46 @@ def update_state(self, entity_id=None, old_state=None, new_state=None):
             self.char_humidity.set_value(humidity, should_callback=False)
             _LOGGER.debug('%s: Percent set to %d%%',
                           self.entity_id, humidity)
+
+
+@TYPES.register('BinarySensor')
+class BinarySensor(HomeAccessory):
+    """Generate a BinarySensor accessory as binary sensor."""
+
+    def __init__(self, hass, state, **kwargs):
+        """Initialize a BinarySensor accessory object."""
+        entity_id = state.entity_id
+        super().__init__(state.name, entity_id, CATEGORY_SENSOR, **kwargs)
+
+        self.hass = hass
+        self.entity_id = entity_id
+
+        service_map = {
+            'gas': ('CarbonMonoxideSensor', 'CarbonMonoxideDetected'),
+            'co2': ('CarbonDioxideSensor', 'CarbonDioxideDetected'),
+            'occupancy': ('OccupancySensor', 'OccupancyDetected'),
+            'opening': ('ContactSensor', 'ContactSensorState'),
+            'motion': ('MotionSensor', 'MotionDetected'),
+            'moisture': ('LeakSensor', 'LeakDetected'),
+            'smoke': ('SmokeSensor', 'SmokeDetected')}
```

Comment:
Those should be constants defined in `./const.py`. E.g.:
 - `DEVICE_CLASS_GAS = 'gas'`
 - `SERV_GAS_SENSOR = CarbonMonoxideSensor`
 - `CHAR_GAS_DETECTED = CarbonMonoxideDetected`
 - ...

`service_map` a constant in `type_sensors.py` outside any class, like so:
```py
SERVICE_MAP = {
    DEVICE_CLASS_GAS: {SERV_GAS_SENSOR, CHAR_GAS_DETECTED),
    # ...
}
```

Reply:
Good coding style. I'll follow it.